enhancements/rhcos/rhcos-cloud-marketplace-images.md

```diff
@@ -0,0 +1,330 @@
+
+Machine pools in the install config would be updated to allow users to more simply
+opt in to using marketplace images without specifying particular image details.
+
+#### Azure
+
+Currently in Azure, to use paid marketplace images, [users consult
+documentation to determine image details](https://docs.openshift.com/container-platform/4.16/installing/installing_azure/installing-azure-customizations.html#installation-azure-marketplace-subscribe_installing-azure-customizations)
+and enter image details in the install config like this:
+
+```yaml
+compute:
+- name: worker
+  platform:
+    azure:
+      osImage:
+        publisher: RedHat
+        offer: rh-ocp-worker
+        sku: rh-ocp-worker
+        version: 413.92.2023101700
+```
+
+A `marketplaceImage` field would be added (OPEN QUESTION: either as a child of `osImage` or as a sibling)
+with the following enum values:
+
+* `NoPurchasePlan` - equivalent to a standard Azure OpenShift install, except under the
+hood we switch the use of the managed image with a non-paid OpenShift image
+* `PurchasePlanNorthAmerica` - equivalent to the yaml snippet directly preceding this
+* `PurchasePlanEMEA` - for users in EMEA, selects marketplace images from the `redhat-limited` publisher
+
+See implementation details below for how these install config values would be translated into selecting
+images from the RHCOS stream.
+
+#### AWS
+
+TODO
+
+#### GCP
+
+TODO
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+ROSA and ARO both make use of marketplace images. ARO HCP depend on the
+availability of Azure marketplace images. It is an Open Question as to whether
+ARO requires a set of Azure marketplace images that is distinct from typical
+Azure OpenShift images.
+
+#### Standalone Clusters
+
+Standalone Azure clusters will have a faster installation time as they will no
+longer be required to upload a VHD with which to create a managed image.
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### RHCOS Stream
+
+The RHCOS stream, [a json file in the installer repo](https://github.com/openshift/installer/blob/master/data/data/coreos/rhcos.json), is the source
+of truth `openshift-install` uses to select & specify versioned first-boot images
+when provisioning machines. The stream contains a top-level `.architectures`
+object and each architecture contains:
+
+* `artifacts`: platform-specific details regarding the source artifact, such
+as file format, file location, SHA, etc.
+* `images`: details about images in specific cloud platforms, particularly
+region to image mappings
+* `rhel-coreos-extensions` (optional): is an extensible object, where the
+only current value is `azure-disk`
+
+#### Azure
+
+In Azure, OpenShift currently publishes *paid* marketplace images
+(which are not published in the RHCOS stream). So we need to add an
+additional free marketplace image, as well as publishing those in
+the stream.
```

Comment:
We need to investigate to which extent this is possible as so far, using an image from a Marketplace listing required additional steps by the user (find the listing, accept the offer, etc). Also, so far all images have been set up with an hourly vCPU price, so we need to investigate if free images are actually possible in Azure Marketplace today.

In summary, this would only be a viable option of the image is usable by any user on Azure without prior steps and without any cost incurred.

Reply:
Non-paid images do not need to accept terms. This enhancement handles discovery of the image for users by including in the stream.